src/mmw/apps/modeling/views.py

```diff
@@ -276,13 +276,16 @@ def _construct_tr55_job_chain(model_input, job_id):
 @decorators.api_view(['GET'])
 @decorators.permission_classes((AllowAny, ))
 def boundary_layer_detail(request, table_code, obj_id):
-    layers = [layer for layer in settings.LAYERS
-              if layer.get('code') == table_code]
-    table_name = layers[0]['table_name']
-    json_field = layers[0].get('json_field', 'geom')
-
-    query = 'SELECT {field} FROM {table} WHERE id = %s'.format(
-            field=json_field, table=table_name)
+    try:
+        layers = [layer for layer in settings.LAYERS
+                  if layer.get('code') == table_code]
+        table_name = layers[0]['table_name']
+        json_field = layers[0].get('json_field', 'geom')
+
+        query = 'SELECT {field} FROM {table} WHERE id = %s'.format(
+                field=json_field, table=table_name)
+    except KeyError:
+        return Response(status=status.HTTP_404_NOT_FOUND)
```

Comment:
I think it might be useful to distinguish between the two failure scenarios: a request to a non-existent resource and not actually specifying a resource.  Consider checking if `table_code` is not provided and returning early with a `400` in that case.